gnucash/gnome/gnc-split-reg.c

```diff
@@ -1305,6 +1305,9 @@ gsr_default_delete_handler( GNCSplitReg *gsr, gpointer data )
     trans = xaccSplitGetParent(split);
     cursor_class = gnc_split_register_get_current_cursor_class (reg);
 
+    /* test for blank_split reference pointing to split */
+    gnc_split_register_change_blank_split_ref (reg, split);
+
```

Comment:
It would be better to separate the test from the action so that this line becomes
```C
    if (gnc_split_register_is_blank_split (reg, split))
        gnc_split_register_change_blank_split_ref (reg, split);
```
Or change the name of the function to something that expresses the real outcome, maybe ```gnc_split_register_ensure_not_blank_split_ref (reg, split)```.

Reply:
Changed this to the first option as it makes more sense.